app/src/main/java/app/grapheneos/apps/ui/MainActivity.kt

```diff
@@ -35,6 +36,16 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         val views = MainActivityBinding.inflate(layoutInflater)
         this.views = views
+        //get the current intent
+        val intent = intent
+
+        //get the attached extras from the intent
+        //we should use the same key as we used to attach the data.
+        val isSuW = intent.getBooleanExtra("SuW", false)
```

Comment:
`SuW` should be declared as a constant

Reply:
should use sud (GlifLayout) if being invoked in setup flow.

Reply:
declared isSuW as constant, currently working on the theme controls

Reply:
You cannot apply const to the variable because if you do the build will fail

Reply:
> should use setupdesign (GlifLayout) if being invoked in setup flow.

how am i meant to use setupdesign because "Apps" is compiled with gradle and setupdesign only works with apps that are compiled using soong.

Reply:
> They meant the intent extra itself "SuW", not `isSuw` needs to be declared as a constant. It's also preferable to use package name as namespace for extras, something like `app.grapheneos.apps.EXTRA_SUW`.

done